### ISSUE TITLE
How is defaultLocale supposed to work?

### ISSUE THREAD

**Issue by buschtoens** [yahoo/ember-intl#280]
It seems that there was a lot of rearranging around `defaultLocale`. Is it supposed to work or not? Because it appears to be broken in the current release (2.0.1).

```js
// config/environment.js
ENV.intl = {
  disablePolyfill: true, // loaded via Polyfill.io
  defaultLocale: 'en-us'
};
```

The locales `en-us` and `de-de` exist. I would expect ember-intl to implicitly chose `en-us`, if no locale was explicitly set.

However, the app doesn't start. I have to explicitly call `intl.setLocal('en-us')` in my application route model hook, otherwise I get this error:

```
Uncaught TypeError: Cannot read property 'join' of null
missingMessage           @	missing-message.js:6
findTranslationByKey     @	intl.js:184
getValue                 @	format-message.js:48
compute                  @	-format-base.js:72
...
```

I'm almost certain that it had worked in a previous release.

**Comment by jasonmit** [comment]
> It seems that there was a lot of rearranging around defaultLocale. Is it supposed to work or not? 

I'm unsure what you mean by rearranging.  Are you referring to #265?

I never really meant for defaultLocale to be the runtime default but the build time default, what the other translations get merged with to handle missing keys.

**Comment by buschtoens** [comment]
> I'm unsure what you mean by rearranging. Are you referring to #265?

Yes, I was.

> I never really meant for defaultLocale to be the runtime default but the build time default [...]

Apparently my memory really is *that* bad. Sorry. :smile:

**Comment by jasonmit** [comment]
Feel free to keep it opened, if the API or bugs are misleading then I consider it a bug.

**Comment by jasonmit** [comment]
#281 should address as much as I understand of the problem.  Let me know if the wording could be more clear.

**Comment by buschtoens** [comment]
Perfect! Thanks :+1:

**Comment by jasonmit** [comment]
Runtime warning fix is released as part of 2.2.1.  Disregard the minor version semver leap, I am starting to lockstep all the deps together now now that all the packages are in the same repository.
